frontend/server/src/test/java/com/amazonaws/ml/mms/ModelServerTest.java

```diff
@@ -273,70 +281,112 @@ private void testLoadModelWithInitialWorkers(Channel channel) throws Interrupted
+    private void testInitialModelDefaultBatchSettings(Channel channel) throws InterruptedException {
+        DescribeModelResponse response = describeModel(channel, "noop-batch");
+
+        Assert.assertEquals(64, response.getBatchSize());
+        Assert.assertEquals(300, response.getMaxBatchDelay());
+    }
+
+    private void testLoadModelWithBatchSizeOverride(Channel channel) throws InterruptedException {
+        unregisterModel(channel, "noop-batch");
+
         result = null;
         latch = new CountDownLatch(1);
         HttpRequest req =
                 new DefaultFullHttpRequest(
-                        HttpVersion.HTTP_1_1, HttpMethod.PUT, "/models/noop_v0.1?min_worker=2");
+                        HttpVersion.HTTP_1_1,
+                        HttpMethod.POST,
+                        "/models?url=noop-batch-v1.0&batch_size=128&max_batch_delay=250");
         channel.writeAndFlush(req);
         latch.await();
 
         StatusResponse resp = JsonUtils.GSON.fromJson(result, StatusResponse.class);
-        Assert.assertEquals(resp.getStatus(), "Processing worker updates...");
+        Assert.assertEquals(resp.getStatus(), "Model \"noop-batch\" registered");
+
+        DescribeModelResponse describeResp = describeModel(channel, "noop-batch");
+
+        Assert.assertEquals(128, describeResp.getBatchSize());
+        Assert.assertEquals(250, describeResp.getMaxBatchDelay());
     }
 
-    private void testSyncScaleModel(Channel channel) throws InterruptedException {
+    private void testLoadModelWithBatchSize(Channel channel) throws InterruptedException {
+        testUnregisterModel(channel);
         result = null;
         latch = new CountDownLatch(1);
         HttpRequest req =
                 new DefaultFullHttpRequest(
                         HttpVersion.HTTP_1_1,
-                        HttpMethod.PUT,
-                        "/models/noop_v0.1?synchronous=true&min_worker=1");
+                        HttpMethod.POST,
+                        "/models?url=noop-v0.1&model_name=noop_v0.1&runtime=python&batch_size=256&max_batch_delay=500");
         channel.writeAndFlush(req);
         latch.await();
 
         StatusResponse resp = JsonUtils.GSON.fromJson(result, StatusResponse.class);
-        Assert.assertEquals(resp.getStatus(), "Workers scaled");
+        Assert.assertEquals(resp.getStatus(), "Model \"noop_v0.1\" registered");
+
+        result = null;
+        latch = new CountDownLatch(1);
+        HttpRequest describeReq =
+                new DefaultFullHttpRequest(
+                        HttpVersion.HTTP_1_1, HttpMethod.GET, "/models/noop_v0.1");
+        channel.writeAndFlush(describeReq);
+        latch.await();
+
+        DescribeModelResponse describeResp = describeModel(channel, "noop_v0.1");
+
+        Assert.assertEquals(256, describeResp.getBatchSize());
+        Assert.assertEquals(500, describeResp.getMaxBatchDelay());
     }
 
```

Comment:
The diff here is sort of messed up for some reason. These methods weren't touched, except for `testUnregisterModel` and `testDescribeModel` - both of which logics has been extracted for reuse in other tests.